initial mcp resource support

joongi007 opened the pull request:
## Changes
This PR adds functionality to convert MCP resources to LangChain Document objects.

### Key Features
- Added `resources.py` module: MCP resources → LangChain Document conversion functionality
- Resource loading capability with single URI or URI list
- Support for text resources with full content and binary resources (stored as metadata with type indication)
- Added `get_resources` method to `MultiServerMCPClient`

### Testing
- Added unit tests: Tests for conversion functions and loading functions

## Checklist
- [x] Added/modified tests
- [x] Run code lint

vbarda commented on langchain_mcp_adapters/resources.py:
```diff
@@ -0,0 +1,93 @@
+    if isinstance(contents, TextResourceContents):
+        content = contents.text
+    elif isinstance(contents, BlobResourceContents):
+        metadata["blob_data"] = contents.blob
```
i agree with @eyurtsev - i think we should be using `Blob` here. would you mind updating?

joongi007 replied:
I've implemented the requested changes. Updated the code to use Blob objects instead of Document for handling binary data, and adjusted the related processing logic accordingly.

vbarda commented on langchain_mcp_adapters/resources.py:
```diff
@@ -0,0 +1,93 @@
+import base64
+
+from langchain_core.documents import Document
+from langchain_core.documents.base import Blob as LCBlob
```
why do we need an alias?

vbarda commented on langchain_mcp_adapters/resources.py:
```diff
@@ -0,0 +1,93 @@
+def convert_mcp_resource_to_langchain_document(
+    resource_uri: str,
+    contents: ResourceContents,
+) -> Document | LCBlob:
```
let's just return a list of `Blob`, no need for a mixed type here, same for the other functions

vbarda commented on langchain_mcp_adapters/resources.py:
```diff
@@ -0,0 +1,93 @@
+            resource_docs = await get_mcp_resource(session, uri)
+            documents.extend(resource_docs)
+        except Exception as e:
+            print(f"Error fetching resource {uri}: {e}")
```
remove print statements, you can re-raise with better error message

vbarda commented:
@joongi007 thanks for updating! looks good overall, but no need to mix `Document` and `Blob` - let's use `Blob` everywhere

joongi007 commented:
@vbarda Thanks for the feedback! I've updated the code to use Blob everywhere and removed the Document usage. Additionally, I’ve improved the error handling by removing print statements and adding more informative error messages.
